type_int32.go

```diff
@@ -26,11 +26,7 @@ func (i *int32PlainDecoder) decodeValues(dst []interface{}) (int, error) {
 		if err := binary.Read(i.r, binary.LittleEndian, &d); err != nil {
 			return idx, err
 		}
-		if i.unSigned {
```

Comment:
This seems like the wrong place to do type coercion. Based on what I understand of the code right now the pattern is to convert the type at the edges of client interaction (write/scan) and reason with the code in parquet types everywhere else. This branching is moving some of that type conversion to a lower level. I believe we should only be reasoning with int32 here.